mleap-spark/src/main/scala/org/apache/spark/ml/bundle/ops/feature/StringIndexerOp.scala

```diff
@@ -17,20 +17,36 @@ class StringIndexerOp extends SimpleSparkOp[StringIndexerModel] {
 
     override def store(model: Model, obj: StringIndexerModel)
                       (implicit context: BundleContext[SparkBundleContext]): Model = {
-      model.withValue("labels", Value.stringList(obj.labels)).
+      var result = model.
+        withValue("labels_length", Value.int(obj.labelsArray.length)).
         withValue("handle_invalid", Value.string(obj.getHandleInvalid))
+      obj.labelsArray.indices.foreach(
+        i => result = result.withValue(s"labels_array_$i", Value.stringList(obj.labelsArray(i)))
+      )
+      result
     }
 
     override def load(model: Model)
```

Comment:
Does the new deserialization logic get tested anywhere?
Do we have a good way to ensure backwards compatibility?

Reply:
The new deserialization logic does get tested by the spark parity tests, though it would only have a labelsArray of length one atm.  I'm working on adding a test with a multi-column input/output for the affected transformers, which would test a longer labelsArray.

Testing that the backwards compatibility works is trickier.  I think the only way would be to serialize an artifact from a spark 2 environment, and then check it into the mleap repo?  @ancasarb thoughts?